src/_main_/utils/common.py

```diff
@@ -170,6 +172,8 @@ def extract_location(args):
     "state": args.pop("state", None),
     "zipcode": args.pop("zipcode", None),
     "country": args.pop("country", 'US'),
+    "building": args.pop("building", None),
+    "room": args.pop("room", None),
```

Comment:
This is duplicate code, which is not used anywhere.
I suggest either using extract_location from within parse_location, or eliminating extract_location